### DISCUSSION TITLE
Could not find a clinic state related to ServiceRequest for a laboratory order

### DISCUSSION THREAD

**Discussion by mduna** [synthetichealth/synthea#1081] [Q&A]
Is it possible to create a ServiceRequest resource for a non-medication order, such as a laboratory order? Thanks.

**Comment by jawalonoski** [comment]
Today, out of the box, it is not possible because Synthea does not currently support the `ServiceRequest` resource.

You have two options: post-process or modify code.

## Post Process

Write a script that processes every patient file, and creates a `ServiceRequest` every time you see a `DiagnosticReport` with the category `LAB`.

## Modify Code

If you are willing and able to modify Java code, you could edit the one of the FHIR exporter classes to produce a `ServiceRequest` prior to every occassion a `DiagnosticReport` is created.

For example, in FHIR R4, here are the two spots of interest where the `DiagnosticReport`s are created:

https://github.com/synthetichealth/synthea/blob/d90b4e3b1c41b3f64af0f4d0518ed6358a01be35/src/main/java/org/mitre/synthea/export/FhirR4.java#L308-L310

https://github.com/synthetichealth/synthea/blob/d90b4e3b1c41b3f64af0f4d0518ed6358a01be35/src/main/java/org/mitre/synthea/export/FhirR4.java#L2229-L2267

**Reply by mduna** [reply]
Thank you for the suggestions, will give a try.
